docs/explanation/planning.rst

```diff
@@ -2,36 +2,67 @@
 Planning
 ========
 
-The plan has a tree structure with three main sections: pre-upgrade steps,
-control-plane upgrade and data-plane upgrade.
-
-.. code:: 
-
-    Top-level plan
-    ├── pre-upgrade steps
-    │   ├── verify cloud is in idle state
-    │   └── MySQL backup
-    ├── control-plane upgrade
-    │   ├── principal applications
-    │   |   ├── application 1..
-    │   |   |   ├── pre-upgrade steps
-    |   |   |   ├── upgrade steps
-    │   |   |   └── post-upgrade steps
-    │   |   └── application N
-    │   |       ├── pre-upgrade steps
-    |   |       ├── upgrade steps
-    │   |       └── post-upgrade steps
-    |   └── subordinate applications...
-    ├── data-plane upgrade
+The plan has a tree structure with six main sections:
+
+* cloud pre-upgrade steps,
+* control-plane principal applications upgrade
+* control-plane subordinate applications upgrade
+* data-plane hypervisors upgrade
+* data-plane non-hypervisor principal applications upgrade
+* data-plane subordinate applications upgrade
+* cloud post-upgrade steps.
+
+.. code:: text
+
+    Upgrade cloud from current release to target release
+        cloud pre-upgrade steps
+            Verify cloud is in idle state
+            MySQL backup
+        control-plane principal applications upgrade
+                application 1
+                    pre-upgrade steps
+                    upgrade steps
+                    post-upgrade steps
+                ...
+                application N
+                    ...
+        control-plane subordinate applications upgrade
+                application M
+                    ...
+        data-plane hypervisors upgrade
+                availability zone 1
+                    pre-upgrade steps
+                    upgrade steps
+                        unit 1
+                            unit upgrade steps
+                        ...
+                        unit K
+                            unit upgrade steps
+                    post-upgrade steps
+                ...
+                availability zone N
+                    ...
+        data-plane non-hypervisor principal applications upgrade
+                application P
+                    ...
+        data-plane subordinate applications upgrade
+                application Q
+                    ...
+        cloud post-upgrade steps
+            (if ceph exists) Ensure correctness of 'require-osd-release' option in 'ceph-mon'
     ...
 
-The **pre-upgrades** steps are used to obtain any further information about the
-applications or to verify their state. 
+The **pre-upgrade** steps prepare COU for the upgrade process, which includes
```

Comment:
Suggestion:
- add a paragraph saying that it's also plan for individual upgrade-groups (control-plane, data-plane and hypervisors) that are a subset of the plan above, and point to the `upgrade_group` section